### PR TITLE
feat: limit one video block per parent

### PR DESCRIPTION
**Pull request by mikeallisonJS**
# Description

copilot:summary

- [Link to Basecamp Todo
](https://3.basecamp.com/3105655/buckets/36562350/todos/7252796180)
# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

copilot:walkthrough

### CODE REVIEW

File: apps/api-journeys/src/app/modules/block/video/video.resolver.ts [hunk 1]
```diff
@@ -140,6 +140,18 @@ export class VideoBlockResolver {
...
       }
+
+      const existingVideosOnParent = await tx.block.count({
+        where: { parentBlockId: input.parentBlockId, typename: 'VideoBlock' }
+      })
+      if (existingVideosOnParent > 0)
+        throw new GraphQLError(
+          'Parent block already has an existing video block',
+          {
+            extensions: { code: 'BAD_REQUEST' }
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/video/video.resolver.ts:151]: This is a User input error. Can you change it to reflect that?

File: apps/api-journeys/src/app/modules/block/video/video.resolver.ts [hunk 2]
```diff
@@ -140,6 +140,18 @@ export class VideoBlockResolver {
             tx
           )
       }
+
+      const existingVideosOnParent = await tx.block.count({
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/video/video.resolver.ts:144]: You shouldn't use count as it unnecessarily goes through the whole table.

The better approach is to use findFirst, which is basically a LIMIT 1 on the database, so the database can stop searching for more results after the first hit.

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-journeys/src/app/modules/block/video/video.resolver.ts (hunk 2 above)
Comment (mikeallisonJS) [reply apps/api-journeys/src/app/modules/block/video/video.resolver.ts]: got it

File: apps/api-journeys/src/app/modules/block/video/video.resolver.ts [hunk 3]
```diff
@@ -140,6 +140,18 @@ export class VideoBlockResolver {
...
       }
+
+      const existingVideoOnParent = await tx.block.findFirst({
+        where: { parentBlockId: input.parentBlockId, typename: 'VideoBlock' }
+      })
+      if (existingVideoOnParent != null)
+        throw new GraphQLError(
+          'Parent block already has an existing video block',
+          {
+            extensions: { code: 'USER_INPUT_ERROR' }
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/video/video.resolver.ts:151]:
```suggestion
            extensions: { code: 'BAD_USER_INPUT' }
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**